frontend/src/js/components/PageViewer/VirtualScroll.tsx

```diff
@@ -101,27 +103,39 @@ export const VirtualScroll: FC<VirtualScrollProps> = ({
     });
   }, [pageHeight, totalPages, debouncedSetPageRange]);
 
-  useEffect(() => {
-    // Inform the parent component of the new middle page
-    // This allows it to do useful things such as have a sensible "next" page
-    // to go to for the find hits
-    onMiddlePageChange(pageRange.middle);
-  }, [pageRange.middle, onMiddlePageChange]);
-
   const throttledSetPageRangeFromScrollPosition = useMemo(() => throttle(setPageRangeFromScrollPosition, 75), [setPageRangeFromScrollPosition]);
 
+  // TODO: try just useEffect
   useLayoutEffect(() => {
-    if (viewport?.current && jumpToPage) {
+    if (viewport?.current) {
       const v = viewport.current;
-      const scrollTo = (jumpToPage - 1) * pageHeight;
-      v.scrollTop = scrollTo;
+      if (jumpToPage) {
+        // TODO: page jump should go to middle?
+        const scrollTo = (jumpToPage - 1) * pageHeight;
+        v.scrollTop = scrollTo;
+      }
+      if (jumpToScrollPosition) {
+        const scrollTo = jumpToScrollPosition + (v.clientHeight / 2);
+        v.scrollTop = scrollTo;
+      }
     }
-  }, [pageHeight, jumpToPage]);
+  }, [pageHeight, jumpToPage, jumpToScrollPosition]);
+
+  useLayoutEffect(() => {
+    if (viewport?.current && focusedFindHighlight) {
+      const v = viewport.current;
+      const topOfHighlightPage = (pageHeight * (focusedFindHighlight.pageNumber - 1));
+      // const highlight = topOfHighlightPage + (focusedFindHighlight.firstSpan?.y ?? 0);
+      // v.scrollTop = highlight + (v.clientHeight / 2);
```

Comment:
tidy this up?  `v` also not an ideal variable name (though short lived!)

Reply:
yeah this was the code that was supposed to jump to a point where the highlight was in the middle of the page (instead of just to the top of that page), but it didn't work because I need to multiply by the page `scale` value which I don't have here

have deleted

unfortunately can't call `v` `viewport` because that name is already bound, but I can just keep referring to `viewport.current` if clearer